common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -23,6 +23,58 @@
 log = logging.getLogger(__name__)
 
 
+def _unenroll_entitlement(course_entitlement, course_run_key):
```

Comment:
I know this is a private method, but it wouldn't hurt to wrap this in its own transaction as well in case we want to call it on its own from another method within this file.

According to https://docs.djangoproject.com/en/1.8/topics/db/transactions/#controlling-transactions-explicitly, it is safe to nest transactions. Inner transactions are rolled back if an exception occurs in the outer transaction at a later point of execution.

Reply:
We're using this method on its own here: https://github.com/edx/edx-platform/pull/16934/files#diff-9b73749954d13e21a49b7ea1e0b110a9R268, therefore I think it should be wrapped in a transaction.